src/components/Announcements/AnnouncementModal.vue

```diff
@@ -0,0 +1,115 @@
+<template>
+  <transition name="modal-fade">
+    <div v-show="open" class="modal-backdrop">
+      <div class="modal">
+        <div class="announce-header">
+          <div class="announce-title">{{announcement.title}}</div>
+          <div class="announce-date">{{ toStringDate(announcement.created) }}</div>
+        </div>
+        <div class="announce-body">{{announcement.description}}</div>
+        <div
+            class="close-btn"
+            @click="close"
+            >
+          X
+        </div>
+      </div>
+    </div>
+  </transition>
+</template>
+
+<script>
+import DateUtils from '../../utils/DateUtils';
+
+export default {
+  name: 'AnnouncementModal',
+  props: {
+    open: {
+      type: Boolean,
+      required: true,
+    },
+    announcement: {
+      type: Object,
+    },
+  },
+  methods: {
+    close() {
+      this.$emit('on-close-announcement');
```

Comment:
Another styling thing, this becomes v-on:on-close-..., might be better to call it v-on:close...